docs/reference/models.md

```diff
@@ -0,0 +1,183 @@
+# Models
+
+Here are described some important relations between our models.
+You can also take a look at [core models](https://github.com/openfun/joanie/tree/main/src/backend/joanie/core/models)
+and [payment models](https://github.com/openfun/joanie/tree/main/src/backend/joanie/payment/models.py).
+
+## Relations between `Order` / `Product` / `Course` / `Course Run` and `Enrollment` models
+
+```mermaid
+erDiagram
+
+    %% Models
+        User {
+            Integer id PK
+        }        
+        Enrollment {
+            User user_id FK
+            CourseRun course_run_id FK
+        }
+        Order {
+            Integer id PK
+            User owner FK
+            Course course FK
+            OrderCourseRelation target_courses FK
+        }
+        CourseRun {
+            string resource_link PK
+        }
+        Course {
+            string code PK 
+        }
+        Product {
+            Integer id PK
+            Course course FK
+            ProductCourseRelation target_courses FK
+        }
+        
+        ProductCourseRelation {
+            Product product FK
+            Course course FK
+            PositiveSmallInteger position
+            Boolean is_graded
+            CourseRun course_runs FK
+        }
+        
+        OrderCourseRelation {
+            Order order FK
+            Course course FK
+            PositiveSmallInteger position
+            Boolean is_graded
+            CourseRun course_runs FK
+        }
+        
+    
+    %% Relations
+        User ||--o{ Enrollment : "has"
+        User ||--o{ Order: "owns"
+        
+        Enrollment }o--|| CourseRun : "is"
+        
+        Order }o--|| Course : ""
+        Order }o--|| Product : ""
+        Order ||--o{ OrderCourseRelation : ""
+        
+        CourseRun }o--|| Course : ""
+        CourseRun }o--o{ OrderCourseRelation : "" 
+        CourseRun }o--|| on_add_course_runs_to_product_course_relation : ""
+        on_add_course_runs_to_product_course_relation || -- o{ ProductCourseRelation : ""
+        
+        Course }o--o{ Product : "displays"
+        Course ||--o{ OrderCourseRelation : "is included in"
+        Course ||--o{ ProductCourseRelation : "is included in"
+        
+        Product ||--o{ ProductCourseRelation : ""
+        
+        ProductCourseRelation ||..|| OrderCourseRelation : "copied at order creation"
+```
+
+### Focus
+
+#### `course` and `target_courses` fields of `Product` and `Order` models
+
+The `course` field represents the course in which the product is sold or the order is owned.
+On the other hand, the `target_courses` field represents the courses contained in the product/order.
```

Comment:
These are the courses that the students will be able to follow once he/she buys the product.